pkg/ipamd/ipamd.go

```diff
@@ -439,14 +439,8 @@ func New(rawK8SClient client.Client, cachedK8SClient client.Client) (*IPAMContex
 	checkpointer := datastore.NewJSONFile(dsBackingStorePath())
 	c.dataStore = datastore.NewDataStore(log, checkpointer, c.enablePrefixDelegation)
 
-	err = c.nodeInit()
-	if err != nil {
-		return nil, err
-	}
-
-	mac := c.awsClient.GetPrimaryENImac()
-
 	// Retrieve security groups
+	mac := c.awsClient.GetPrimaryENImac()
```

Comment:
I looked at this code in CNI 1.12.1
https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.12.1/pkg/ipamd/ipamd.go#L414

It hasn't changed in that. 

- So, I wasn't entirely certain if this is the cause of regression here.
- Secondly, will `c.awsClient.GetPrimaryENImac` return a success if the there a chance of `c.nodeInit` failure?

Reply:
What changed is that node initialization will now attach new ENIs: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.13.0/pkg/ipamd/ipamd.go#L612 instead of just reconciling already attached CNIs: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.12.6/pkg/ipamd/ipamd.go#L589

So the change is that we are attempting ENI allocations before the security groups from the primary ENI are synced, which leads to this error.

For `GetPrimaryENImac`, this will always be populated during container init from IMDS, and the `aws-node` will not be able to start if we cannot find the primary ENI MAC associated with the instance. So there is no behavior change there. It is still possible for `nodeInit` to fail for many reasons, but this failure was a "false positive".

Reply:
Got it. Thanks for the context and explanation.